src/Receipt/Receipt.php

```diff
@@ -32,91 +31,39 @@ class Receipt {
 	 */
 	public $message = '';
 
-	/**
-	 * Donation id.
-	 *
-	 * @since 2.7.0
-	 * @var int $donationId
-	 */
-	private $donationId;
-
 	/**
 	 * Receipt details group class names.
 	 *
 	 * @since 2.7.0
 	 * @var array
 	 */
-	private $detailsGroupList = [
-		DonorDetailsGroup::class,
-		DonationDetailsGroup::class,
-		AdditionalDetailsGroup::class,
-	];
-
-	/**
-	 * Receipt constructor.
-	 *
-	 * @since 2.7.0
-	 * @param $donationId
-	 */
-	public function __construct( $donationId ) {
-		$this->donationId = $donationId;
-
-		$this->setupReceipt();
-
-	}
+	protected $sectionList = [];
 
 	/**
-	 * Setup receipt.
+	 * Get receipt sections.
 	 *
+	 * @return array
 	 * @since 2.7.0
 	 */
-	private function setupReceipt() {
-		/**
-		 * Filter the detail group class names.
-		 *
-		 * Developer can use this filter hook to register there detail group.
-		 */
-		$this->detailsGroupList = apply_filters( 'give_receipt_register_details_group', $this->detailsGroupList, static::class, $this->donationId );
+	public function getSections() {
+		return $this->sectionList;
 	}
 
 	/**
-	 * Get detail group object.
+	 * Add detail group.
 	 *
-	 * @param string $class
+	 * @param  array $section
 	 *
-	 * @return DetailGroup|null
 	 * @since 2.7.0
 	 */
-	public function getDetailGroupObject( $class ) {
-		/**
-		 * Use this filter to handle object initialization for your detail group class.
-		 *
-		 * @since 2.7.0
-		 */
-		$object = apply_filters( 'give_receipt_create_detail_group_object', null, $class, $this->donationId );
-
-		if ( $object instanceof DetailGroup ) {
-			return $object;
-		}
-
-		$classNames           = $this->getDetailGroupList();
-		$detailGroupClassName = $classNames[ array_search( $class, $classNames, true ) ];
-
-		if ( $detailGroupClassName ) {
-			$object = new $detailGroupClassName( $this->donationId );
-		}
-
-		return $object;
-	}
+	abstract public function addSection( $section );
 
 	/**
-	 * Return list of detail group class names.
+	 * Remove receipt section.
 	 *
-	 * @since 2.7.0
+	 * @param  string $sectionId
 	 *
-	 * @return array
+	 * @since 2.7.0
 	 */
-	public function getDetailGroupList() {
-		return $this->detailsGroupList;
-	}
+	abstract public function removeSection( $sectionId );
```

Comment:
Just define the method in here, no need to force the child class to define it.

Reply:
Make sense.

Reply:
fixed: https://github.com/impress-org/givewp/compare/ccb59a12b626...55c57984a9f5